virtcontainers/network.go

```diff
@@ -1138,36 +1264,65 @@ func createVirtualNetworkEndpoint(idx int, ifName string, interworkingModel NetI
 		return &VirtualEndpoint{}, fmt.Errorf("invalid network endpoint index: %d", idx)
 	}
 
-	uniqueID := uuid.Generate().String()
-
-	hardAddr := net.HardwareAddr{0x02, 0x00, 0xCA, 0xFE, byte(idx >> 8), byte(idx)}
+	netPair := createNetworkInterfacePair(idx, ifName, interworkingModel)
 
 	endpoint := &VirtualEndpoint{
 		// TODO This is too specific. We may need to create multiple
 		// end point types here and then decide how to connect them
 		// at the time of hypervisor attach and not here
-		NetPair: NetworkInterfacePair{
-			ID:   uniqueID,
-			Name: fmt.Sprintf("br%d", idx),
-			VirtIface: NetworkInterface{
-				Name:     fmt.Sprintf("eth%d", idx),
-				HardAddr: hardAddr.String(),
-			},
-			TAPIface: NetworkInterface{
-				Name: fmt.Sprintf("tap%d", idx),
-			},
-			NetInterworkingModel: interworkingModel,
-		},
+		NetPair:      netPair,
 		EndpointType: VirtualEndpointType,
 	}
+	if ifName != "" {
+		endpoint.NetPair.VirtIface.Name = ifName
+	}
+
+	return endpoint, nil
+}
+
+func createBridgedMacvlanNetworkEndpoint(idx int, ifName string, interworkingModel NetInterworkingModel) (*BridgedMacvlanEndpoint, error) {
+	if idx < 0 {
+		return &BridgedMacvlanEndpoint{}, fmt.Errorf("invalid network endpoint index: %d", idx)
+	}
+
+	netPair := createNetworkInterfacePair(idx, ifName, interworkingModel)
 
+	endpoint := &BridgedMacvlanEndpoint{
+		// TODO This is too specific. We may need to create multiple
+		// end point types here and then decide how to connect them
+		// at the time of hypervisor attach and not here
+		NetPair:      netPair,
+		EndpointType: BridgedMacvlanEndpointType,
+	}
 	if ifName != "" {
 		endpoint.NetPair.VirtIface.Name = ifName
 	}
 
 	return endpoint, nil
 }
 
+func createNetworkInterfacePair(idx int, ifName string, interworkingModel NetInterworkingModel) NetworkInterfacePair {
+	uniqueID := uuid.Generate().String()
+
+	hardAddr := net.HardwareAddr{0x02, 0x00, 0xCA, 0xFE, byte(idx >> 8), byte(idx)}
```

Comment:
OOI, why are we using these specific values?

In other news, I now feel strangely in need of a coffee... :smile:

Reply:
@jodh-intel  :)
I have just refactored this code, moved this part of existing code into a new function.
Anyways I will add comments in here to explain.

Also, I was planning to have this code merged after the network hotplug PR lands..